---
title: "R Notebook - SNP data"
author: "Ian Brettell"
output: html_notebook
---

*8 December 2017*

Purpose: To determine whether the SNPs that are in LD with those targeted fall within (or without) the same genes

Initial setup

```{r, eval = FALSE}

install.packages("tidyverse")

source("https://bioconductor.org/biocLite.R")
biocLite()
biocLite("biomaRt")

biocLite("GenomicRanges")
library(GenomicRanges)
biocLite("Homo.sapiens")
library(Homo.sapiens)
biocLite("stephenturner/annotables")
library(annotables)
biocLite("ensemblVEP")
library(ensemblVEP)
biocLite("snpStats")

install.packages("MatrixEQTL")
```


## 1.1 Import files

```{r}
aibl_snps <- read.delim("C:/Users/bre227/Dropbox/eQTL/Data/Expanded SNP set/AIBLgene_SNP_LIST_04032015.tsv")
snap_edit <- read.delim("C:/Users/bre227/Dropbox/eQTL/Data/Expanded SNP set/snapResults_edit.txt")
snap <- read.delim("C:/Users/bre227/Dropbox/eQTL/Data/Expanded SNP set/snapResults.txt")
```

## 1.2 Combine datasets

Note: 'snap_edit' has removed all SNPs from 'snap' that did not have proxies.

```{r}
# How many SNPs are there in snap_edit?
length(unique(snap_edit$SNP)) # Shows that 557 have been removed from aibl_snps. 
# How many genes are there in aibl_snps?
length(unique(aibl_snps$GENE))
# How many aibl_snps don't have an associated gene?
which(aibl_snps$GENE == "")
# One entry has "-" in the gene column.
which(aibl_snps$GENE == "-") # shows that there is 1 additional SNP with no associated gene
length(setdiff(aibl_snps$SNP, snap_edit$SNP)) # same number as above - 557 removed.
# How many unique proxy SNPs?
length(unique(snap_edit$Proxy))
```

```{r}
library(tidyverse)
AIBLdummy <- dplyr::select(aibl_snps, SNP, GENE) # creates dummy data frame with just two variables
snap2 <- left_join(snap_edit, AIBLdummy, by = "SNP") # combine data frames
rm(AIBLdummy) # remove dummy
snap2 <- dplyr::select(snap2, SNP, SNP_gene = "GENE", dplyr::everything())# reorder columns
```

*19 December 2017*

## 1.3 Get gene names and locations for AIBL_snps

Create vector of aibl_snps

```{r}
snps <- as.vector(aibl_snps$SNP)
```

Get ensembl ids and loci

```{r, results = "hide"}
library(biomaRt)
listMarts()
mart_snps <- useMart('ENSEMBL_MART_SNP')
listDatasets(mart_snps) # lists datasets available
mart_snps <- useMart('ENSEMBL_MART_SNP', 'hsapiens_snp')
```

```{r}
SNP_genes <- getBM(attributes = c("refsnp_id", "allele_1", "minor_allele", "minor_allele_freq", "chr_name", "chrom_start", "chrom_end", "ensembl_gene_stable_id"),
                     filters = "snp_filter",
                     values = snps,
                     mart = mart_snps)
```

See how many SNPs returned results
```{r}
length(unique(SNP_genes$refsnp_id))
```

Create `ensembl_ids' vector of unique ensembl gene ids from 'SNP_genes'

```{r}
ensembl_ids <- unique(SNP_genes$ensembl_gene_stable_id)
```

Use 'ensembl_ids' to get loci and hgnc symbols for genes

```{r, results = "hide"}
listMarts()
mart_genes <- useMart('ENSEMBL_MART_ENSEMBL')
listDatasets(mart_genes)
mart_genes <- useMart('ENSEMBL_MART_ENSEMBL', 'hsapiens_gene_ensembl')
```

```{r}
ensembl_genes <- getBM(attributes = c("ensembl_gene_id", "chromosome_name", "start_position", "end_position", "strand", "hgnc_symbol", "entrezgene"),
                       filters = "ensembl_gene_id",
                       values = ensembl_ids,
                       mart = mart_genes)
```

Bind both tables by ensembl_id to give full table of genes associated with the AIBL snps

```{r}
SNP_genes_full <- left_join(SNP_genes, ensembl_genes, by = c("ensembl_gene_stable_id" = "ensembl_gene_id"))
```

<!---
This tests whether we can remove the rows that returned "LRG..." in the 'ensemble_gene_stable_id' column without losing any information.
```{r}
test <- SNP_genes_full[grep("LRG", SNP_genes_full$ensembl_gene_stable_id), ] # creates data frame with just "LRG" rows (161 rows total)
test2 <- SNP_genes_full[-grep("LRG", SNP_genes_full$ensembl_gene_stable_id), ] # creates data frame with no "LRG" rows
setdiff(test$refsnp_id, test2$refsnp_id) # tests whether there are any differences between the refsnp_ids of 'test' and 'test2' (that is, if there are differences, the "LRG" rows would not be duplicates). Returns 'character(0)'
which((test$refsnp_id %in% test2$refsnp_id) == "FALSE") # just to make sure, we test again whether there were any non-matches. Returns 'integer(0), i.e. all refsnp_ids are in both 'test' and 'test2'.
rm(test)
rm(test2)
```
--->

Remove duplicate rows with "LRG..." in the 'ensembl_gene_stable_id' column

```{r}
SNP_genes_full <- SNP_genes_full[-grep("LRG", SNP_genes_full$ensembl_gene_stable_id), ]
dim(SNP_genes_full)
length(unique(SNP_genes_full$refsnp_id))
length(unique(SNP_genes_full$ensembl_gene_stable_id))

# SNP_genes_full$ensembl_gene_stable_id has blank entries - replace with 'NA'
SNP_genes_full$ensembl_gene_stable_id[SNP_genes_full$ensembl_gene_stable_id == ""] <- NA
length(unique(SNP_genes_full$refsnp_id[!is.na(SNP_genes_full$ensembl_gene_stable_id)]))

# write table
write.table(SNP_genes_full, "C:/Users/bre227/Documents/R/AD_project/Working/aibl_snp_genes.txt", sep = "\t", col.names = T, row.names = F, quote = F)

```


## 1.4 Get gene names and locations for proxy snps

Create vector of proxy snps

```{r}
proxy_snps <- as.vector(unique(snap_edit$Proxy))
length(proxy_snps)
```

Get ensembl ids and loci for proxies

```{r, results = "hide"}
library(biomaRt)
listMarts()
mart_snps <- useMart('ENSEMBL_MART_SNP')
listDatasets(mart_snps) # lists datasets available
mart_snps <- useMart('ENSEMBL_MART_SNP', 'hsapiens_snp')

proxy_genes <- getBM(attributes = c("refsnp_id", "chr_name", "chrom_start", "chrom_end", "ensembl_gene_stable_id"),
                     filters = "snp_filter",
                     values = proxy_snps,
                     mart = mart_snps)
```


See how many SNPs returned results loci information
```{r}
length(unique(proxy_genes$refsnp_id))
```


<!-- Again, testing whether we can remove the rows that returned "LRG..." in the 'ensemble_gene_stable_id' column without losing any information.
```{r}
test <- proxy_genes[grep("LRG", proxy_genes$ensembl_gene_stable_id), ] # creates data frame with just "LRG" rows (161 rows total)
test2 <- proxy_genes[-grep("LRG", proxy_genes$ensembl_gene_stable_id), ] # creates data frame with no "LRG" rows
setdiff(test$refsnp_id, test2$refsnp_id) # tests whether there are any differences between the refsnp_ids of 'test' and 'test2' (that is, if there are differences, the "LRG" rows would not be duplicates). Returns 'character(0)'
which((test$refsnp_id %in% test2$refsnp_id) == "FALSE") # just to make sure, we test again whether there were any non-matches. Returns 'integer(0), i.e. all refsnp_ids are in both 'test' and 'test2'.
rm(test)
rm(test2)
```
--->

Remove rows with "LRG" in 'ensemble_gene_stable_id'

```{r}
proxy_genes <- proxy_genes[-grep("LRG", proxy_genes$ensembl_gene_stable_id), ]
dim(proxy_genes)
```

Create `ensembl_ids' vector of unique ensembl gene ids from 'proxy_genes'

```{r}
ensembl_ids_proxies <- unique(proxy_genes$ensembl_gene_stable_id)
length(ensembl_ids_proxies)
```

Use 'ensembl_ids' to get loci and hgnc symbols for genes

```{r, results = "hide"}
listMarts()
mart_genes <- useMart('ENSEMBL_MART_ENSEMBL')
listDatasets(mart_genes)
mart_genes <- useMart('ENSEMBL_MART_ENSEMBL', 'hsapiens_gene_ensembl')
```

```{r, results = "hide"}
ensembl_proxy_genes <- getBM(attributes = c("ensembl_gene_id", "chromosome_name", "start_position", "end_position", "strand", "hgnc_symbol"),
                       filters = "ensembl_gene_id",
                       values = ensembl_ids_proxies,
                       mart = mart_genes)
```

Bind both tables by ensembl_id to give full table of genes associated with the AIBL snps

```{r}
proxy_genes_full <- left_join(proxy_genes, ensembl_proxy_genes, by = c("ensembl_gene_stable_id" = "ensembl_gene_id"))
dim(proxy_genes_full)

# write table

write.table(proxy)
```

## 1.5 Determine differences between genes in SNP and proxy lists

Find out how many unique genes there are in the SNP and proxy lists

```{r}
length(unique(SNP_genes_full$ensembl_gene_stable_id))
length(unique(proxy_genes_full$ensembl_gene_stable_id))
```


Find out how many proxy genes are different from the SNP genes, and vice versa

```{r}
length(setdiff(SNP_genes_full$ensembl_gene_stable_id, proxy_genes_full$ensembl_gene_stable_id))
length(setdiff(proxy_genes_full$ensembl_gene_stable_id, SNP_genes_full$ensembl_gene_stable_id))
```
So 126/621 genes in the AIBL SNP list  are not in the proxy list, and 608/1103 genes in the proxy list are not in the SNP list.

*22 December 2017*

## g:Profiler test

Used the g:Profiler tool (<https://biit.cs.ut.ee/gprofiler/>) to see whether it could get the location of some of the 9 SNPs that BiomaRt couldn't detect.

Pull out those 9 SNP ids

```{r}
setdiff(aibl_snps$SNP, SNP_genes_full$refsnp_id)
```

Write text file

```{r}
write.table(setdiff(aibl_snps$SNP, SNP_genes_full$refsnp_id), 
            file = "C:/Users/bre227/Documents/R/AD_project/Working/missing_aibl_snps.txt",
            quote = F,
            col.names = F,
            row.names = F)
```

Copy and paste text of file into g:SNPense, and then read output.

```{r}
gout <- read.csv("C:/Users/bre227/Documents/R/AD_Project/Working/missing_aibl_snps_gprofiler_output.csv",
         header = T,
         sep = "\t")
```

Provided details for two of the nine missing SNPs. To test whether it works for all the SNPs that were identified with BioMart:

```{r}
write.table(as.vector(aibl_snps$SNP),
            file = "C:/Users/bre227/Documents/R/AD_project/Working/aibl_snps.txt",
            quote = F,
            col.names = F,
            row.names = F)

gout2 <- read.csv("C:/Users/bre227/Documents/R/AD_Project/Working/aibl_snps_gprofiler_output.csv",
         header = T,
         sep = "\t")
```

For how many SNPs can g:Profiler not find location information?

```{r}
# replace blanks with NA
gout2$chr_pos[gout2$chr_pos == "Not found"] <- NA 
length(which(is.na(gout2$chr_pos)))
```

This suggests that the best approach would be to integrate the two methods (biomaRt and g:Profiler).

***

*8 January 2018*

### Read in genotype data

```{r}
library(readxl)
gtypes <- read_excel("C:/Users/bre227/Dropbox/eQTL/Data/Expanded SNP set/AIBLgene_SNP_Data_by_Gene_Expression_IDs.xlsx",
                col_names = F)
```

Note that the AIBL IDs are incorrect in the excel file. We attach the correct IDs using the 'AIBL_Gene_Expression_IDs_UpdtdDec2017.txt' file, **assuming that both sets of IDs, when sorted, correspond with each other**.

```{r}
ids2 <- read.delim("C:/Users/bre227/Dropbox/eQTL/Data/AIBL_expression_set/AIBL_Gene_Expression_IDs_UpdtdDec2017.txt", sep = " ")

# bind colnames for exdata to new IDs
ids2$old_ids <- as.integer(gsub("X", "", colnames(exdata)))

# reorder to same order as the SNP genotype data
ids2 <- ids2[order(ids2$old_ids), ]

# merge with rownames of genotype data to ensure perfect match
ids3 <- data.frame(gtypes[5:nrow(gtypes), 1])# extract AIBL IDs from genotype data
ids3$X__1 <- as.integer(ids3$X__1)
ids3 <- merge(ids3, ids2, by.x = "X__1", by.y = "old_ids")

# add new column names to the genotype data
ids2 <- as.vector(ids3$x)
gtypes$AIBL_ID_new <- c(NA, NA, NA, "AIBL_ID_new", as.vector(ids2))
gtypes <- dplyr::select(gtypes, AIBL_ID_new, dplyr::everything()) # reorder to bring AIBL_ID_new to the front
```


```{r, results = "hide"}
library(tidyverse)
gts <- as.tibble(t(gtypes)) # transpose data frame
gts <- as.tibble(lapply(gts, function(x) { # convert all 'X_X' to NA
  gsub("X_X", NA, x)
}))
```

Because some individuals failed to record a result for some SNPs in addition to the few individuals for whom no results were recorded, we will manually remove the latter.
```{r}
colnames(gts) <- gts[3, ] # make the column names the (incorrect) AILB IDs

# some individuals have data missing for ~1,500 of the ~2,000 SNPs, e.g.

length(which(is.na(gts$`127`)) == TRUE)

x <- sapply(gts, function(x) { # gets TRUE or FALSE for each column with data missing for more than 1000 SNPs
  length(which(is.na(x) == TRUE)) > 1000
})
y <- which(x == TRUE) # creates vector of column indexes for which it is true
gts[y] <- NULL # removes those columns

colnames(gts) <- c("chromosome", "position", "gene", "snp", gts[1, 5:ncol(gts)]) # replace column names with correct AIBL IDs

gts <- gts[-c(1:3), -c(1:3)] # remove extraneous first three columns and first three rows
```

Write table to working folder

```{r}
write.table(gts, file = "C:/Users/bre227/Documents/R/AD_project/Working/genotype_data.txt", sep = "\t")
```

*12 January 2018*

## Reformat data to PED and MAP for analysis

In accordance with the guide here: <http://zzz.bwh.harvard.edu/plink/data.shtml>

<!--
Import data and create map file
```{r}
ped <- read.table("C:/Users/bre227/Documents/R/AD_project/Working/genotype_data.txt", header = T)
map <- data.frame(ped[, 1:4], stringsAsFactors = F)
map <- dplyr::select(map, chromosome, snp, position)
map$chromosome <- as.character(map$chromosome)
map$snp <- as.character(map$snp)
# check that there are no unexpected values
unique(map$chromosome)
# noticed that there is an empty row - remove from map and ped file
map <- map[-which(is.na(map$chromosome)), ]
ped <- ped[-which(is.na(ped$chromosome)), ]
# replace chr "X" with "23" as required by format
map$chromosome[map$chromosome == "X"] <- "23"
```

Find duplicated SNP entries in the PED and determine whether all the genotype data is also duplicated (in which case we can delete).
```{r}
length(unique(map$snp)) # only 2,088 out of 2,150 SNPs are unique?
# get duplicated rs IDs from MAP file
map$snp[which(duplicated(map$snp))]
# create new data frame with genotype data from PED file matching those rs IDs
dupe <- ped[ped$snp %in% map$snp[which(duplicated(map$snp))], ]

# By visualising them, we see that an individual's entries for the duplicated SNP can be different. JD suggested to remove them all.

map <- map[!map$snp %in% dupe$snp, ] # remove from map
ped <- ped[!ped$snp %in% dupe$snp, ] # remove from ped
```

Write map file
```{r}
map <- data.frame(lapply(map, function(x){
  gsub(" ", "", x)
}), stringsAsFactors = F)
write.table(map, "C:/Users/bre227/Documents/R/AD_project/Working/snp_data.map", row.names = F, col.names = F, quote = F, sep = "\t")
```

Reformat ped into PED format
```{r}
ped2 <- ped[, -c(1:4)] # remove extraneous columns
meta <- read.table("C:/Users/bre227/Documents/R/AD_project/Working/key_metadata.txt", header = T) # read in metadata
colnames(ped2) <- gsub("X", "", colnames(ped2)) # remove "X" from colnames
ped2 <- data.frame(t(ped2), stringsAsFactors = F) # transpose ped2
ped2$AIBL.Id <- rownames(ped2) # make a new column with the row names (AIBL Ids) to use to bind with metadata
meta$AIBL.Id <- as.character(meta$AIBL.Id) # convert to characters to allow binding
library(dplyr)
ped2 <- left_join(ped2, meta, by = "AIBL.Id")
ped3 <- select(ped2, AIBL.Id, Demographic.Sex, PET, everything())
ped3[, grep("Age|apoe4", colnames(ped3), ignore.case = T)] <- NULL # remove Age and apoe4 columns


# convert male/female to 1/2
ped3$Demographic.Sex <- as.character(ped3$Demographic.Sex)
ped3$Demographic.Sex[ped3$Demographic.Sex == "Male"] <- "1"
ped3$Demographic.Sex[ped3$Demographic.Sex == "Female"] <- "2"

# convert PET status from POS/NEG to 1/0
ped3$PET <- as.character(ped3$PET)
ped3$PET[ped3$PET == "POS"] <- "2"
ped3$PET[ped3$PET == "NEG"] <- "1"
ped3$PET[is.na(ped3$PET)] <- "0"

# clean data
ped4 <- data.frame(lapply(ped3, function(x){
  gsub(" ", "", x)
  gsub("_", " ", x)
}), stringsAsFactors = F)
ped4[is.na(ped4)] <- "0 0"

# write file
write.table(ped4, "C:/Users/bre227/Documents/R/AD_project/Working/snp_data.ped", row.names = F, col.names = F, quote = F, sep = "\t")
```
-->

*15 January 2018*

Tried to run PLINK, but it came up with an error in the ped file. Find out what's wrong:

```{r}
ped <- read.table("C:/Users/bre227/Documents/R/AD_project/Working/genotype_data.txt", header = T)

# find unique values across genotype data
as.character(unique(unlist(ped[, 5:ncol(ped)])))

# shows that there are some with values "Homozygous Allele 2/Allele 2", II", "DI", and "DD". They must have caused the errors. Find out which rows they are in
df <- t(apply(ped, 1, function(x) grepl("II|DI|DD|Homozygous Allele 2/Allele 2", x[5:length(x)])))
df2 <- which(df == "TRUE", arr.ind = T)
unique(df2[, 1]) # to get the rows (genes) that contain the above strings
ped <- ped[-unique(df2[, 1]), ]# remove those rows
# noticed that the row names are not re-adjusted, so we'll do that manually
rownames(ped) <- seq(1:nrow(ped))

# PLINK revealed another error - AIBL.Id 284 has a third allele for rs2247856 - convert to NA
a <- grep("rs2247856", ped$snp)
b <- grep("X284", colnames(ped))
ped[a, b] <- NA

# After trying to run it again, PLINK revealed a further error - we need to find which rows (SNPs) have all NA and remove
na_gns <- apply(ped[, 5:ncol(ped)], 1, function(x) length(which(is.na(x) == T)) == length(5:ncol(ped)))
length(which(na_gns == T)) # shows that 40 SNPs have no data
ped <- ped[-which(na_gns == T), ] # remove
```

Repeat process above...

Create map file

```{r}
map <- data.frame(ped[, 1:4], stringsAsFactors = F)
map <- dplyr::select(map, chromosome, snp, position)
map$chromosome <- as.character(map$chromosome)
map$snp <- as.character(map$snp)
# check that there are no unexpected values
unique(map$chromosome)
# noticed that there is an empty row - remove from map and ped file
map <- map[-which(is.na(map$chromosome)), ]
ped <- ped[-which(is.na(ped$chromosome)), ]
# replace chr "X" with "23" as required by format
map$chromosome[map$chromosome == "X"] <- "23"
unique(map$chromosome)
```

Find duplicated SNP entries in the PED and determine whether all the genotype data is also duplicated (in which case we can delete).
```{r}
length(unique(map$snp)) # now only 2,085 out of 2,150 SNPs are unique
# get duplicated rs IDs from MAP file
map$snp[which(duplicated(map$snp))]
# create new data frame with genotype data from PED file matching those rs IDs
dupe <- ped[ped$snp %in% map$snp[which(duplicated(map$snp))], ]

# By visualising them, we see that an individual's entries for the duplicated SNP can be different. JD suggested to remove them all.

map <- map[!map$snp %in% dupe$snp, ] # remove from map
ped <- ped[!ped$snp %in% dupe$snp, ] # remove from ped
```

Write map file
```{r}
map <- data.frame(lapply(map, function(x){
  gsub(" ", "", x)
}), stringsAsFactors = F)
write.table(map, "C:/Users/bre227/Documents/R/AD_project/Working/plink/snp_data.map", row.names = F, col.names = F, quote = F, sep = "\t")
```

Reformat ped into PED format
```{r}
ped2 <- ped[, -c(1:4)] # remove extraneous columns
meta <- read.table("C:/Users/bre227/Documents/R/AD_project/Working/key_metadata.txt", header = T) # read in metadata
colnames(ped2) <- gsub("X", "", colnames(ped2)) # remove "X" from colnames
ped2 <- data.frame(t(ped2), stringsAsFactors = F) # transpose ped2
ped2$AIBL.Id <- rownames(ped2) # make a new column with the row names (AIBL Ids) to use to bind with metadata
meta$AIBL.Id <- as.character(meta$AIBL.Id) # convert to characters to allow binding
library(dplyr)
ped2 <- left_join(ped2, meta, by = "AIBL.Id")
ped3 <- dplyr::select(ped2, AIBL.Id, Demographic.Sex, PET, everything())
ped3[, grep("Age|apoe4", colnames(ped3), ignore.case = T)] <- NULL # remove Age and apoe4 columns


# convert male/female to 1/2
ped3$Demographic.Sex <- as.character(ped3$Demographic.Sex)
ped3$Demographic.Sex[ped3$Demographic.Sex == "Male"] <- "1"
ped3$Demographic.Sex[ped3$Demographic.Sex == "Female"] <- "2"

# convert PET status from POS/NEG to 1/0
ped3$PET <- as.character(ped3$PET)
ped3$PET[ped3$PET == "POS"] <- "2"
ped3$PET[ped3$PET == "NEG"] <- "1"
ped3$PET[is.na(ped3$PET)] <- "0"

# clean data
ped4 <- data.frame(lapply(ped3, function(x){
  gsub(" ", "", x)
  gsub("_", " ", x)
}), stringsAsFactors = F)
ped4[is.na(ped4)] <- "0 0"

# write file
write.table(ped4, "C:/Users/bre227/Documents/R/AD_project/Working/plink/snp_data.ped", row.names = F, col.names = F, quote = F, sep = "\t")
```

I fed the files to PLINK using the following code: `./plink --file snp_data --map3 --noweb --no-fid --no-parents --assoc` and `./plink --file snp_data --map3 --noweb --no-fid --no-parents --fisher`, which created the output files 'plink.assoc' and 'plink.assoc.fisher'. 

Take a look at the files.

```{r}
pk_ass <- read.table("C:/Users/bre227/Documents/R/AD_project/Working/plink/plink.assoc", na.strings = NA, header = T, fill = T)
pk_fsh <- read.table("C:/Users/bre227/Documents/R/AD_project/Working/plink/plink.assoc.fisher", na.strings = NA, header = T, fill = T)

length(which(pk_ass$P < 0.05))
length(which(pk_fsh$P < 0.05))
```

Shows that the Fisher exact test output had fewer significant SNPs, so we'll prefer that output.

```{r}
#create table of significant SNPs
sig_pk <- pk_fsh[(pk_fsh$P < 0.05) == T &!is.na(pk_fsh$P), ]
dim(sig_pk)
# merge with gene ids from 'aible_snp_genes.txt' file
snp_gns <- read.delim("C:/Users/bre227/Documents/R/AD_project/Working/aibl_snp_genes.txt", sep = "\t", header = T)
library(dplyr)
sig_pk <- left_join(sig_pk, snp_gns, by = c("SNP" = "refsnp_id"))
# note that some SNPs map to multiple genes

# how many unique genes?
length(unique(sig_pk$hgnc_symbol))
length(unique(sig_pk$ensembl_gene_stable_id))
length(unique(sig_pk$entrezgene))

# order by P value
sig_pk <- sig_pk[order(sig_pk$P), ]

# rename and reorder columns
sig_pk <- dplyr::select(sig_pk, 
                 SNP, 
                 minor_allele = "A1", 
                 freq_in_cases = "F_A", 
                 freq_in_controls = "F_U", 
                 major_allele = "A2", 
                 p_value = "P", 
                 odds_ratio_minor_allele = "OR", 
                 chr = "chr_name", 
                 snp_locus = "chrom_start", 
                 gene_start = "start_position", 
                 gene_end = "end_position", 
                 strand, ensembl_id = "ensembl_gene_stable_id", 
                 hgnc_symbol, entrez_id = "entrezgene")

# write table
write.table(sig_pk, "C:/Users/bre227/Documents/R/AD_project/Working/sig_PET_pos_v_neg_alleles_annotated.txt", col.names = T, row.names = F, quote = F, sep = "\t")
```

Combine data with probe id (by entrez id, which is the most annotated of gene ids)
```{r}
probes <- read.delim("C:/Users/bre227/Documents/R/AD_project/Working/affy_probes_annotated.txt", header = T, sep = "\t")
sig_pk <- left_join(sig_pk, probes, by = "entrez_id")
# remove second ensembl id and gene symbol columns (they have no additional information)
sig_pk <- dplyr::select(sig_pk, -ensembl_id.y, -gene_symbol) %>% 
  rename(ensembl_id.x = "ensembl_id")

# add NAs to hgnc_symbol column
sig_pk$hgnc_symbol[sig_pk$hgnc_symbol == ""] <- NA
sig_pk$hgnc_symbol <- as.character(sig_pk$hgnc_symbol)

# how many genes?
length(unique(sig_pk$ensembl_id))
```

*16 January 2018*

<!--
See if we can get more annotation information by using the locations of the SNPs to get the gene names, rather than the ensembl IDs.

```{r}
library(GenomicRanges)
# include "chr" prefix before each chromosome name
cnv <- SNP_genes
cnv$chr_name <- paste("chr", cnv$chr_name, sep = "")

# make GRanges object for SNPs
cnv1 <- makeGRangesFromDataFrame(cnv,
                                keep.extra.columns = F,
                                seqnames.field = "chr_name",
                                start.field = "chrom_start",
                                end.field = "chrom_end")

# create function to match overlaps
#splitColumnByOverlap <- function(query, subject, column = "ENTREZID", ...){
#    olaps <- findOverlaps(query, subject, ...)
#    f1 <- factor(subjectHits(olaps),
#                 levels=seq_len(subjectLength(olaps)))
#    splitAsList(mcols(query)[[column]][queryHits(olaps)], f1)
#}

# load library
library(Homo.sapiens)
gns <- genes(Homo.sapiens, columns = c("ENTREZID", "SYMBOL", "ENSEMBL"))
head(gns)

# run function
ols <- findOverlaps(gns, cnv1, select = "all")
ols
unique(queryHits(ols)) # only returns hits for 227 SNPs??

ols1 <- subsetByOverlaps(gns, cnv1)
ols1 # confirmed?

ols2 <- data.frame(ols1)

```
-->

*17 January 2018*

**JD**: Run permutations (10k). Alternate / full model association tests. Logistic models with covariates: age, gender, apoe4.

Obtain reference allele from BioMart query, in accordance with: <http://zzz.bwh.harvard.edu/plink/dataman.shtml#refallele> 
```{r}
# read in table
snp_gns <- read.delim("C:/Users/bre227/Documents/R/AD_project/Working/aibl_snp_genes.txt", header = T)
ref_al <- dplyr::select(snp_gns, refsnp_id, minor_allele)
ref_al <- as.tibble(sapply(ref_al, as.character), stringsAsFactors = F)
ref_al <- ref_al[!duplicated(ref_al), ]

write.table(ref_al, "C:/Users/bre227/Documents/R/AD_project/Working/plink/ref_alleles.txt", row.names = F, col.names = F, quote = F, sep = "\t")
```

Write covariate file, in accordance with: <http://zzz.bwh.harvard.edu/plink/data.shtml#covar>
```{r}
cov <- ped2[,(ncol(ped2) - 5 + 1):ncol(ped2)] # to obtain metadata bound to aibl IDs for which we have genotype data
cov$Demographic.Sex <- as.character(cov$Demographic.Sex)
cov$Demographic.Sex[cov$Demographic.Sex == "Male"] <- "1"
cov$Demographic.Sex[cov$Demographic.Sex == "Female"] <- "2"
colnames(cov)[colnames(cov) == "Demographic.Sex"] <- "Sex"
cov$PET <- NULL # because it's already in the PED file
cov$FID <- cov$AIBL.Id # to make a family ID column - see <http://zzz.bwh.harvard.edu/plink/data.shtml#ped>
cov <- dplyr::select(cov, FID, IID = "AIBL.Id", everything())
write.table(cov, "C:/Users/bre227/Documents/R/AD_project/Working/plink/cov.txt", row.names = F, col.names = T, quote = F, sep = "\t")
```

Tried to run `./plink --file snp_data --logistic --genotypic --beta --reference-allele ref_alleles.txt --covar.txt` but it returned the error message: "No file [snp_data.fam] exists."

So we created a .fam file using `./plink --file snp_data --out snp_data --make-bed --map3 --no-fid --no-parents` and tried again.

Output file created (then renamed) as: 'plink.assoc.logistic.genotype'.

Then ran again without the genotype query: `./plink --file snp_data --logistic --beta --reference-allele ref_alleles.txt --covar.txt`

Also re-ran `--assoc` and `--fisher` with the reference alleles file to ensure conformity with the logistic regression output (and then ran the above code again).

Import just the logistic file (without the genoytpe command).

```{r}
log <- read.delim("C:/Users/bre227/Documents/R/AD_project/Working/plink/plink.assoc.logistic", header = T, sep = "")
log <- log[with(log, order(CHR, BP)), ] # sort by chromosome and bp

# Find SNPs that are significant indpendent of covariates
log2 <- dplyr::select(log[!is.na(log$P), ], -NMISS, -BETA, -STAT) %>% 
  tidyr::spread(key = TEST, value = P)

length(which(log2$ADD < 0.05))
length(which(log2$Age < 0.05))
length(which(log2$apoe4 < 0.05))
length(which(log2$Sex < 0.05))
```

This means there apoe4 was a significant covariate in *all* SNPs! (As was age in all but two)

Let's pull out the SNPs with ADD < 0.05 and annotate them.

```{r}
# pull out SNPs that are significant
sig_snps_log <- log[log$SNP %in% log2$SNP[which(log2$ADD < 0.05)] & log$TEST == "ADD", ]
# attach annotation data
sig_snps_log <- dplyr::left_join(sig_snps_log, snp_gns, by = c("SNP" = "refsnp_id"))
```

# **NOTE** redo association test with permutation before binding the sig_snps_log to the assoc output

*18 January 2018*

See whether any genes cited in Apostolova et al. (2018) *Associations of the top 20 ALheimer Disease risk variants with brain amyloidosis* are in our significant SNPs list.

```{r}
# first check how many hits we get with these genes from the original SNPs list
snp_gns[grep("PICALM|BIN1|CR1|ABCA7|MS4A6A|CLU|FERMT2|DSG2|EPHA1|SORL1|ZCWPW1", snp_gns$hgnc_symbol), ]
nrow(snp_gns[grep("PICALM|BIN1|CR1|ABCA7|MS4A6A|CLU|FERMT2|DSG2|EPHA1|SORL1|ZCWPW1", snp_gns$hgnc_symbol), ])

# now check how many hits we get with our significant SNPs list
sig_snps_log[grep("PICALM|BIN1|CR1|ABCA7|MS4A6A|CLU|FERMT2|DSG2|EPHA1|SORL1|ZCWPW1", sig_snps_log$hgnc_symbol), ]
```
Only one - CLU.

Ran PLINK again for association, but this time with 10,000 permutations and the reference allele file: `./plink --file snp_data --map3 --noweb --no-fid --no-parents --assoc --mperm 10000 --reference-allele ref_alleles.txt`. Output two files:

* plink.assoc
* plink.assoc.mperm (with the permutation results)

Take a look again, and compare to the logistic regression output

```{r}
ass <- read.table("C:/Users/bre227/Documents/R/AD_project/Working/plink/plink.assoc", na.strings = NA, header = T, fill = T)
sig_ass <- ass[ass$P < 0.05 & is.na(ass$P) == F, ]

perm <- read.table("C:/Users/bre227/Documents/R/AD_project/Working/plink/plink.assoc.mperm", na.strings = NA, header = T, fill = T)
perm[which(perm$EMP2 < 0.7), ]
```

Only two SNPs have a corrected p-value of less than 0.7! And neither is significant.

***

*19 January 2018*

Use Haploview to see if we get the same permutation results as PLINK.

```{r}
y <- dplyr::select(snp_gns, refsnp_id, chr_name, chrom_start)
y$loc <- paste(y$chr_name, y$chrom_start, sep = "")
y$loc <- paste(y$chr_name, y$chrom_start, sep = ":")
y$chr_name <- NULL
y$chr_start <- NULL
y <- y[y$refsnp_id %in% map$snp, ]
write.table(y, "C:/Users/bre227/Documents/R/AD_project/Working/haploview/snp_locs.info", col.names = F, row.names = F, sep = "\t", quote = F)


ped5 <- ped4
ped5$family_id <- ped5$AIBL.Id
ped5$paternal_id <- 0
ped5$maternal_id <- 0
ped5 <- dplyr::select(ped5, family_id, AIBL.Id, paternal_id, maternal_id, Demographic.Sex, PET, everything())

write.table(ped5, "C:/Users/bre227/Documents/R/AD_project/Working/haploview/snp_data.ped", col.names = F, row.names = F, sep = "\t", quote = F)
```

Got the same results with an association test with 10,000 permutations - only two SNPs had p < 0.7, and neither was significant.

## eQTL analysis

Read in data.

```{r}
exdata <- read.delim("C:/Users/bre227/Dropbox/eQTL/Data/AIBL_expression_set/AIBL_Gene_Expression.txt", sep = "", header = T)
ids2 <- read.delim("C:/Users/bre227/Dropbox/eQTL/Data/AIBL_expression_set/AIBL_Gene_Expression_IDs_UpdtdDec2017.txt", sep = " ")
colnames(exdata) <- as.vector(ids2$x) # replace colnames with ids2
meta <- read.delim("C:/Users/bre227/Documents/R/AD_project/Working/key_metadata.txt", header = T, sep = "\t")
```

### Put into format required by `MatrixEQTL` package.

#### Covariates file

```{r}
library(tidyverse)

# create binary statuses for sex and PET
meta$Demographic.Sex <- ifelse(meta$Demographic.Sex == "Male", 0, 
                               ifelse(meta$Demographic.Sex == "Female", 1, NA))
meta$PET <- ifelse(meta$PET == "POS", 1,
                   ifelse(meta$PET == "NEG", 0, NA))

# tidy metadata
meta <- data.frame(t(meta), stringsAsFactors = F)
colnames(meta) <- meta[1, ]
meta$id <- rownames(meta)
rownames(meta) <- seq(1:nrow(meta))
meta <- dplyr::select(meta, id, dplyr::everything())
meta <- meta[-1, ]

# tidy probe data
exdata$probe_id <- rownames(exdata)
exdata <- dplyr::select(exdata, probe_id, dplyr::everything())
rownames(exdata) <- seq(1:nrow(exdata))
```

#### Genotype file

```{r}

# tidy genotype data
ped6 <- ped[, -c(1:3)]
ped6 <- left_join(ped6, dplyr::select(snp_gns[!duplicated(snp_gns$refsnp_id), ], refsnp_id, allele_1, minor_allele), by = c("snp" = "refsnp_id")) %>% 
  select(snp, allele_1, minor_allele, everything())
ped6 <- data.frame(lapply(ped6, function(x) gsub("_", " ", x)),
                   stringsAsFactors = F)
```

Noticed that for some SNPs the alleles annotated with BioMart differ from the alleles present in our data (e.g. "rs1064261" is annotated as having A and G alleles, but our data shows T and C), and some SNPs are annotated as having the same major and minor alleles (e.g. "rs3753584" which has C for both its major and minor allele). 

We will therefore get another plink output to find calculate which alleles are major and minor when they differ from the BioMart annotations, or when the BioMart annotations seem incorrect. We used the code: `./plink --file snp_data --map3 --noweb --no-fid --no-parents --fisher --out allele_freq`

```{r}
# import plink data on allele frequencies
pk_frq <- read.table("C:/Users/bre227/Documents/R/AD_project/Working/plink/allele_freq.assoc.fisher", na.strings = NA, header = T, fill = T)
pk_frq <- dplyr::select(pk_frq, snp = SNP, pk_major_allele = A2, pk_minor_allele = A1)

# create comparison data frame
comp <- data.frame(left_join(dplyr::select(ped6, snp, allele_1, minor_allele), pk_frq, by = "snp") %>% 
  select(snp, allele_1, pk_major_allele, minor_allele, pk_minor_allele), stringsAsFactors = F)
comp <- data.frame(sapply(comp, as.character), stringsAsFactors = F)

# create empty columns for correct allele calls
comp$cor_mj <- NA
comp$cor_mn <- NA

# substitute empty values with NA
comp[, 2:5] <- sapply(comp[, 2:5], function(x) gsub("0", NA, x))
comp[, 2:5] <- sapply(comp[, 2:5], function(x) gsub("^$|^ $", NA, x))

# add correct major allele column
comp$cor_mj <- unlist(apply(comp, 1, function(x){
  if (is.na(x[2]) == T & is.na(x[3]) == T){ # if both are NA, return NA
    x[6] <- NA
  } else if (x[2] == x[5] & x[4] == x[3] & is.na(x[2] == F) & is.na(x[3] == F)){ # if annotated and plink alleles are swapped around, prefer annotated alleles
    x[6] <- x[2]  
  } else if (is.na(x[2]) == T & is.na(x[3] == F)){ # if only plink output has an allele, use that
    x[6] <- x[3]
  } else if (is.na(x[2]) == F & is.na(x[3] == T)){ # if only the annotation has an allele, use that
    x[6] <- x[2]
  } else if (x[2] == x[3] & is.na(x[2] == F) & is.na(x[3] == F)) { # if the annotated and plink alleles are the same, just use the annotated allele
    x[6] <- x[2]
  } else { # otherwise use the plink allele
    x[6] <- x[3]
  }
}))

# add correct minor allele column
comp$cor_mn <- unlist(apply(comp, 1, function(x){
    if (is.na(x[4]) == T & is.na(x[5]) == T){
    x[7] <- NA
  } else if (x[2] == x[5] & x[4] == x[3] & is.na(x[2] == F) & is.na(x[3] == F)){
    x[7] <- x[4]
  } else if (is.na(x[4]) == T & is.na(x[5] == F)){
    x[7] <- x[5]
  } else if (is.na(x[4]) == F & is.na(x[5] == T)){
    x[7] <- x[4]
  } else if (x[4] == x[5] & is.na(x[4]) == F & is.na(x[5]) == F) {
    x[7] <- x[4]
  } else {
    x[7] <- x[5]
  }
}))
```

*23 January 2018*

Incorporate adjusted major and minor allele into ped6 and create genotype where:
*0 = homozygous major allele
*1 = heterozygous
*2 = homozygous minor allele
```{r}
ped6$allele_1 <- comp$cor_mj
ped6$minor_allele <- comp$cor_mn

#

test <- ped6[1:5, ]

out <- t(data.frame(apply(test, 1, function(x){ ## WORKS
  a <- paste(x[2], x[2], sep = " ")
  b <- paste(x[2], x[3], sep = " ")
  c <- paste(x[3], x[2], sep = " ")
  d <- paste(x[3], x[3], sep = " ")
  results <- sapply(x, function(y){
    if (is.na(y) == T){
      y <- NA
    }
    else if (y == a & is.na(y) == F){
      y <- 0
    }
    else if (y == b | y == c & is.na(y) == F){
      y <- 1
    } 
    else if (y == d & is.na(y) == F){
      y <- 2
    }
  })
  data.frame(do.call(rbind, results))
})))

out <- t(data.frame(apply(ped6, 1, function(x){ ## WORKS - need to change the second function from sapply to lapply
  a <- paste(x[2], x[2], sep = " ")
  b <- paste(x[2], x[3], sep = " ")
  c <- paste(x[3], x[2], sep = " ")
  d <- paste(x[3], x[3], sep = " ")
  results <- lapply(x[4:length(x)], function(y){
    if (is.na(y) == T){
      y <- NA
    }
    else if (y == a & is.na(y) == F){
      y <- 0
    }
    else if (y == b | y == c & is.na(y) == F){
      y <- 1
    } 
    else if (y == d & is.na(y) == F){
      y <- 2
    }
  })
  data.frame(do.call(rbind, results))
})))

# bind snp info to genotype coding
ped7 <- data.frame(cbind(ped6[, 1], out), stringsAsFactors = F)
rownames(ped7) <- seq(1:nrow(ped7))
ped7 <- dplyr::rename(ped7, snpid = V1)
colnames(ped7) <- gsub("X", "", colnames(ped7))
```

#### Expression file

```{r}
exdata <- read.delim("C:/Users/bre227/Dropbox/eQTL/Data/AIBL_expression_set/AIBL_Gene_Expression.txt", sep = "", header = T)
ids2 <- read.delim("C:/Users/bre227/Dropbox/eQTL/Data/AIBL_expression_set/AIBL_Gene_Expression_IDs_UpdtdDec2017.txt", sep = " ")
colnames(exdata) <- as.vector(ids2$x) # replace colnames with ids2

# extract only those columns (samples) for which we have genotype data
exdata <- exdata[, colnames(exdata) %in% colnames(ped7)]
```

#### Finalise covariates file by extracting only those columns (samples) for which we have genotype data

```{r}
cns <- c("id", colnames(meta)[colnames(meta) %in% colnames(ped7)])
meta <- meta[, cns]
```


<!--ignore below

## 1.3 Get annotation for proxy SNPs

Create vector with unique SNPs

```{r}
uq_proxies <- as.vector(unique(snap$Proxy)) 
```

Access SNP mart

```{r}
library(biomaRt)
listMarts() # lists marts available
mart <- useMart('ENSEMBL_MART_SNP')
listDatasets(mart) # lists datasets available
mart <- useMart('ENSEMBL_MART_SNP', 'hsapiens_snp')
```

Get loci for SNPs

```{r}
proxy_loci <- getBM(attributes = c("refsnp_id", "chr_name", "chrom_start", "chrom_end"),
                    filters = c("snp_filter"),
                    values = c(uq_proxies),
                    mart = mart)
```



<!--Ignore below

create vectors for start and end loci
proxy_start <- as.vector(proxy_loci$chrom_start)
proxy_end <- as.vector(proxy_loci$chrom_end)

 access gene mart
listMarts()
gene_mart <- useMart('ENSEMBL_MART_ENSEMBL')
listDatasets(gene_mart)
gene_mart <- useMart('ENSEMBL_MART_ENSEMBL', 'hsapiens_gene_ensembl')

get genes associated with proxy SNPs
proxy_gns <- getBM(attributes = c("start_position", "hgnc_symbol"),
                   #filters = c("start", "end"),
                   #values = list(proxy_start, proxy_end),
                   #mart = gene_mart) # query: command takes four hours to run?

output was 49 million observations of 2 variables??



# Alternative method to the one above

genelist <- grch38 %>% 
  dplyr::select(symbol, chr, start, end)
gns <- makeGRangesFromDataFrame(genelist,
                                keep.extra.columns = T,
                                ignore.strand = T,
                                seqnames.field = "chr",
                                start.field = "start",
                                end.field = "end")

proxygr <- makeGRangesFromDataFrame(proxy_loci,
                                    ignore.strand = T,
                                    seqnames.field = "chr_name",
                                    start.field = "chrom_start",
                                    end.field = "chrom_end")

overlapGenes <- data.frame(findOverlaps(proxygr, gns))
regionsWIthHits <- data.frame(proxygr[overlapGenes$queryHits])
regionsWIthHits$genes <- gns$symbol[overlapGenes$subjectHits]
length(unique(regionsWIthHits$genes)) # returns 734


## test using online BioMart

#mart <- useMart('ENSEMBL_MART_SNP', 'hsapiens_snp')
#proxy_loci <- getBM(attributes = c("refsnp_id", "chr_name", "chrom_start", "chrom_end", "chrom_strand"),
                    #filters = c("snp_filter"),
                    #values = c(uq_proxies),
                    #mart = mart) # showed that all ~18,000 SNPs lie on the forward strand??

#proxies_500 <- head(snap2, n = 500) %>% 
#  dplyr::select(Proxy, Chromosome, Coordinate_HG18)
#colnames(proxies_500)[colnames(proxies_500) == "Coordinate_HG18"] <- "Start"
#proxies_500$Strand <- 1
#chr <- as.character(proxies_500$Chromosome)
#chr <- gsub("chr", "", chr)
#proxies_500$Chromosome <- chr
#rm(chr)
#proxies_500$loci <- paste(proxies_500$Chromosome, proxies_500$Start, proxies_500$Start, "1", sep = ":")
#write.table(proxies_500$loci, file = "biomart.test.txt", quote = F, row.names = F, col.names = F)

## only returned a handfull of entries, instead of the expectd 500

##################################################

# test using ensemblVEP package

library(ensemblVEP)
write.table(uq_proxies, "~/R/AD Project/Working/uq_proxies.txt", quote = F, row.names = F, col.names = F)
## submit file to "http://www.ensembl.org/Tools/VEP" and use output as follows

uq_proxies_output <- read.delim("~/R/AD Project/Working/uq_proxies_VEP_output.txt", header = T)

# extract unique SNPid and gene names
prxy_snpsandgns <- dplyr::select(uq_proxies_output, X.Uploaded_variation, SYMBOL)
uq_prxy_snpsandgns <- prxy_snpsandgns[!duplicated(prxy_snpsandgns), ]

# produce same file but for AIBL SNPs

AIBL_SNPs <- as.vector(unique(aibl_snps$SNP))
write.table(AIBL_SNPs, "~/R/AD Project/Working/AIBL_SNPs.txt", quote = F, row.names = F, col.names = F)

# extract unique SNPid and gene names

aibl_snps_output <- read.delim("~/R/AD Project/Working/AIBL_SNPs_VEP_output.txt", header = T)
AIBL_snpsandgns <- dplyr::select(aibl_snps_output, X.Uploaded_variation, SYMBOL)
uq_AIBL_snpsandgns <- AIBL_snpsandgns[!duplicated(AIBL_snpsandgns), ]

########################

# through bioconductor
library(ensemblVEP)
myparam <- VEPFlags(version = max(unlist(currentVEP())),
                    scriptPath = character(),
                    flags <- list(vcf = F, 
                                 everything,
                                 output_file = "~/R/AD Project/Working/AIBL_SNPs.txt"))

-->